encord/orm/dataset.py

```diff
@@ -54,19 +58,93 @@ class DatasetUsers:
     pass
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
 class DataClientMetadata:
     payload: dict
 
 
+@dataclasses.dataclass(frozen=True)
+class SignedUrl:
+    signed_url: str
+
+
+@dataclasses.dataclass(frozen=True)
+class ImageData:
+    id: int
+    title: str
+    user_hash: UUID
+    file_link: str
```

Comment:
Can we make this `file_link` the permanent file link that is in our database and have another field `signed_url` which is the signed url? So we can have both?